content-resources/src/main/java/flyway/oskari/V3_1_1__remove_analysis_from_appsetups.java

```diff
@@ -0,0 +1,205 @@
+package flyway.oskari;
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class V3_1_1__remove_analysis_from_appsetups extends BaseJavaMigration {
+    private static final Logger LOG = LogFactory.getLogger(flyway.oskari.V3_0_1__infobox_colourscheme_to_theme.class);
+    private static final String ANALYSIS_LAYER_PLUGIN_ID = "Oskari.mapframework.bundle.mapanalysis.plugin.AnalysisLayerPlugin";
+    private static final String ANALYSIS_LAYER_PREFIX = "analysis_";
+    private static final int BATCH_SIZE = 1000;
+    @Override
+    public void migrate (Context context) throws Exception {
+        Connection conn = context.getConnection();
+        removeAnalysisLayerFromMapfullState(conn);
+        removeAnalysisLayerPluginFromMapmodules(conn);
+        removeAnalyseFromAppsetupBundles(conn);
+        removeAnalyseFromBundles(conn);
+    }
+
+    private void removeAnalysisLayerFromMapfullState(Connection conn) throws SQLException {
+        JSONArray mapfullConfigs = getMapfullConfigsWithAnalysisLayer(conn);
+        String sql = """
+            UPDATE oskari_appsetup_bundles bundles SET state = v.state 
+            FROM (VALUES
+                ${values}
+            ) AS v(appsetup_id, bundle_id, state)
+            WHERE bundles.appsetup_id = v.appsetup_id AND bundles.bundle_id = v.bundle_id
+        """;
+
+        List<String> values = new ArrayList<>();
+        Statement statement = conn.createStatement();
+
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.YYYY HH:mm:ss");
+        LOG.debug(OffsetDateTime.now().format(dtf) + " Starting update.");
+        for (int i = 0, counter = 0; i < mapfullConfigs.length(); i++, counter++) {
+            JSONObject mapfull = mapfullConfigs.optJSONObject(i);
+            JSONObject state = new JSONObject(mapfull.getString("state"));
+
+            if (state == null || !state.has("selectedLayers")) {
```

Comment:
JSONObject state can't be null, but `new JSONObject(mapfull.getString("state"));` could probably throw an exception if `mapfull.getString("state")` is null